projects/step-core/src/lib/modules/entity/components/entity-icon/entity-icon.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Input } from '@angular/core';
+import { EntityScopeResolver } from '../../services/entity-scope-resolver';
+import { Entity } from '../../types/entity';
+import { EntityRegistry } from '../../services/entity-registry';
+
+import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
+import { AJS_MODULE } from '../../../../shared';
+
+@Component({
+  selector: 'entity-icon',
+  templateUrl: './entity-icon.component.html',
+  styleUrls: ['./entity-icon.component.scss'],
+})
+export class EntityIconComponent {
+  @Input() entityName?: string;
+  @Input() entity!: Entity;
+
+  icon?: string;
+  tooltip?: string;
+
+  constructor(private entityScopeResolver: EntityScopeResolver, private entityRegistry: EntityRegistry) {}
+
+  ngOnChanges(): void {
+    this.icon = 'adjust';
```

Comment:
Still required some fixes. `ngOnChanges` is invoked on every change detection cycle. To prevent redundant code executions, we need to check the required input parameters were really changed. Usually I use the following approach:
```
// Define additional method with the main logic of change's reaction. 
// As parameters it should accepts values that reaction is depends on
// In current case there are entity and entityName
// Both parameter are optional
private updateIconAndTooltip(entity?: Entity, entityName?: string): void {

   // If paramter not set, take value from the field
  entity = entity || this.entity;
  entityName = entityName || this.eneityName;

  // Do other stuff, which is based on parameter'values
  ....
}

// In ngOnChanges analyze which parameters were changed
// and invoke the reaction method
ngOnChanges(changes: SimpleChanges): void {

  let entity: Entity | undefined = undefined;
  let entityName: string | undefined = unedfined;

  const cEntity = changes.entity;
  const cEntityName = changes.entityName;

  if (cEntity?.previousValue !== cEntity?.currentValue) {
     entity = cEntity?.currentValue;
  }  

  if (cEntityName?.previousValue !== cEnityName?.currentValue) {
     entityName = cEntityName?.currentValue;
  }

  // entity and entityName variables will be undefined, in case if there is no changes for them 
  // In that case no reaction will be invoked.

  // Otherwise it will be invoked even one of them will be changed.
  if (entity || entityName) {

    // In case if entity or entityName will be empty
    // updateIconAndTooltip will take acutal value from component's field.
    this.updateIconAndTooltip(entity, entityName);  
  }

}
```
I hope the idea is clear :-) Feel free to modify it, if you see how it can be written in more optimal and clear way.

Reply:
I ended up doing something similar. I noticed that for some angularjs components implementing the new ng2+ component for the initial value previousValue and currentValue where the same thus skipping the update function, so I also looked for changes.xyz.firstChange.